domain/agentpassword/state/state.go

```diff
@@ -343,6 +343,89 @@ WHERE  name=$entityName.name`, u)
 	return u.UUID, errors.Capture(err)
 }
 
+// MatchesModelPasswordHash checks if the password is valid or not againt the
+// password hash stored for the model's agent.
+func (s *State) MatchesModelPasswordHash(
+	ctx context.Context, passwordHash agentpassword.PasswordHash,
+) (bool, error) {
+	db, err := s.DB()
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	args := validateModelPasswordHash{PasswordHash: passwordHash}
+	query := `
+SELECT COUNT(*) AS &validateModelPasswordHash.count
+FROM   model_agent
+WHERE  password_hash = $validateModelPasswordHash.password_hash;
+`
+	stmt, err := s.Prepare(query, args)
+	if err != nil {
+		return false, errors.Errorf("preparing statement to set password hash: %w", err)
+	}
+
+	var count int
```

Comment:
We can just use the existing `count` as output and `modelPasswordHash` struct as input.
No need to create a new struct `validateModelPasswordHash` and define this new variable.

Reply:
This is the way  the rest of the domain was done. I think it may have been done to be reentrant safe. Going to leave it for consistency.